pkg/openstacknet/funcs.go

```diff
@@ -85,3 +91,173 @@ func GetOpenStackNetWithLabel(r common.ReconcilerCommon, namespace string, label
 	}
 	return &osNetList.Items[0], nil
 }
+
+// GetOpenStackNetsMapWithLabel - Return a map[NameLower] of all OpenStackNets in the namespace that have (optional) labels
+func GetOpenStackNetsMapWithLabel(
+	r common.ReconcilerCommon,
+	namespace string,
+	labelSelector map[string]string,
+) (map[string]ospdirectorv1beta1.OpenStackNet, error) {
+	osNetList, err := GetOpenStackNetsWithLabel(
+		r,
+		namespace,
+		labelSelector,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	osNetMap := map[string]ospdirectorv1beta1.OpenStackNet{}
+	for _, osNet := range osNetList.Items {
+		osNetMap[osNet.Spec.NameLower] = osNet
+	}
+
+	return osNetMap, nil
+}
+
+//
+// AddOSNetNameLowerLabels - add osnetcfg CR label reference which is used in
+// the in the osnetcfg controller to watch this resource and reconcile
+//
+func AddOSNetNameLowerLabels(
+	r common.ReconcilerCommon,
+	obj client.Object,
+	cond *ospdirectorv1beta1.Condition,
+	networkNameLowerNames []string,
+) error {
+
+	currentLabels := obj.GetLabels()
+	if currentLabels == nil {
+		currentLabels = map[string]string{}
+	}
+
+	osNetLabels := map[string]string{}
+	removedOsNets := map[string]bool{}
+	newOsNets := map[string]bool{}
+	networkNameLowerNamesMap := map[string]bool{}
+
+	for _, n := range networkNameLowerNames {
+		label := fmt.Sprintf("%s/%s", SubNetNameLabelSelector, n)
+		networkNameLowerNamesMap[label] = true
+	}
+
+	//
+	// get current osNet Labels and verify if nets got removed
+	//
+	for _, label := range reflect.ValueOf(currentLabels).MapKeys() {
+		//
+		// has label key SubNetNameLabelSelector string included?
+		//
+		if strings.HasSuffix(label.String(), SubNetNameLabelSelector) {
+			l := label.String()
+			osNetLabels[l] = currentLabels[l]
+
+			//
+			// if l is not in networkNameLowerNamesMap it got removed
+			//
+			if _, ok := networkNameLowerNamesMap[l]; !ok {
+				delete(currentLabels, l)
+				removedOsNets[l] = true
+
+			}
+		}
+	}
+
+	if len(newOsNets) > 0 {
+		common.LogForObject(
+			r,
+			fmt.Sprintf("%s %s removing network labels: %v",
+				obj.GetObjectKind().GroupVersionKind().Kind,
+				obj.GetName(),
+				removedOsNets,
+			),
+			obj,
+		)
+	}
+
+	//
+	// identify if nets got added
+	//
+	for label := range networkNameLowerNamesMap {
+		//
+		// if l is not in networkNameLowerNamesMap it got removed
```

Comment:
Is this comment correct?

Reply:
oh, this is something I stopped working on for now and do in a follow up to be able to remove subnets

Reply:
Ack

Reply:
updated to reflect current state